cmd/kube-controller-manager/app/bootstrap.go

```diff
@@ -20,11 +20,14 @@ import (
 	"context"
 	"fmt"
 
+	"k8s.io/klog/v2"
+
 	"k8s.io/controller-manager/controller"
 	"k8s.io/kubernetes/pkg/controller/bootstrap"
 )
 
 func startBootstrapSignerController(ctx context.Context, controllerContext ControllerContext) (controller.Interface, bool, error) {
+	ctx = klog.NewContext(ctx, klog.LoggerWithName(klog.FromContext(ctx), "bootstrap-signer-controller"))
```

Comment:
would it make more sense for the caller to wrap the context to get contextual logging before calling the start___Controller method, so the thing setting up all the controller loops could guarantee this was done consistently, and could do so using the identifier already associated with each controller loop (like the admin-facing initializer name)?

Reply:
That makes sense. I suppose that would be in [StartControllers](https://github.com/kubernetes/kubernetes/blob/eb4e2a2225bd534aa7d765510fbf9e441d498c5e/cmd/kube-controller-manager/app/controllermanager.go#L566-L567).

Is the initializer name the string set in  [NewControllerInitializers](https://github.com/kubernetes/kubernetes/blob/eb4e2a2225bd534aa7d765510fbf9e441d498c5e/cmd/kube-controller-manager/app/controllermanager.go#LL423C5-L423C31)?

@atiratree: my preference would be to not ask all the different contributors to revise their PRs, but instead merge them when the rest of the changes are okay. Then once they are merged, I can prepare a PR which removes the individual names from all controller and instead does it in StartControllers. It will be less work overall.

Reply:
that is a good idea - to move the responsibility for consistency outside

@pohly yup, doing it in one sweep sounds good (as we are still alpha)

Reply:
and yes we would use the initializer names set (after we make it consistent)

Reply:
> That makes sense. I suppose that would be in [StartControllers](https://github.com/kubernetes/kubernetes/blob/eb4e2a2225bd534aa7d765510fbf9e441d498c5e/cmd/kube-controller-manager/app/controllermanager.go#L566-L567).

Exactly

> Is the initializer name the string set in [NewControllerInitializers](https://github.com/kubernetes/kubernetes/blob/eb4e2a2225bd534aa7d765510fbf9e441d498c5e/cmd/kube-controller-manager/app/controllermanager.go#LL423C5-L423C31)?

Yes

> @atiratree: my preference would be to not ask all the different contributors to revise their PRs, but instead merge them when the rest of the changes are okay. Then once they are merged, I can prepare a PR which removes the individual names from all controller and instead does it in StartControllers. It will be less work overall.

If we haven't merged the changes to put context logging construction inside controllers yet, it seems questionable to proceed adding that to all the controllers only to then remove it.

Reply:
I avoids extra roundtrips with contributors who may or may not react to further change requests in a timely manner.  It's of course cleaner to not add these lines in the first place.

Reply:
Discussed this with @soltysh and we agreed to move ahead with these PRs as they are and finish up later.

/lgtm